VMware.CloudFoundation.Reporting.psm1

```diff
@@ -824,7 +824,158 @@ Function Request-SoSHealthJson {
+            } else {
+                $reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")
+            }
+        }
+        if (Test-VCFConnection -server $server) {
+            # Use Rest API method to request for health summary
+            $uri = "https://$server/v1/system/health-summary"
+            $healthSummaryPayload = ConvertTo-JSON $healthSummarySpec -depth 10
+            $response = Invoke-RestMethod -Method POST -URI $uri -headers $vcfHeaders -body $healthSummaryPayload
+			if ($response.id -eq "") {
+				write-error "Health Summary request run into issues please rerun the script"
+				return $false
+			}
+			$requestID = $response.id
+			
+            # Pulling request status
+            $uri = "https://$server/v1/system/health-summary/$requestID"
+            $response = Invoke-RestMethod -Method GET -URI $uri -headers $vcfHeaders
+            $escapeCounter = 0
+			while (($escapeCounter -lt 30) -and !($response.status -match "COMPLETED")) {
+				sleep(30)
+				$escapeCounter++
+				$response = Invoke-RestMethod -Method GET -URI $uri -headers $vcfHeaders
+			}
+			if ($escapeCounter -eq 20) {
+				write-error "Health Summary request is taking an unusal amount of time, please check SDDC Manager UI for any issues"
+				return $false
+			}
+
+			# Create a temporary directory under reportDirectory
+            $createPathCounter = 0
+			for ($createPathCounter -lt 4) {
+				$randomOutput = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 6 |%{[char]$_})
+				$outFilePath = $reportPath + "\" + $randomOutput
+				if (!(Test-Path -Path $outFilePath) -and (Test-Path -Path $reportPath)) {
+					break
+				} else {
+					if ($createPathCounter -eq 3) {
+						write-host "unable to write to $reportPath"
+					}
+					$createPathCounter++
+				}
+			}
+			New-Item -Path $outFilePath -ItemType Directory | Out-NULL
+
+            # Downloading summary tar.gz to temporary directory
+            $uri = "https://$server/v1/system/health-summary/$requestID/data"
+			$outFile = $outFilePath + "\health-summary.tar.gz"
+            $response = Invoke-RestMethod -Method GET -URI $uri -headers $vcfHeaders -OutFile $outFile
+
+            # Unzip/untar tar.gz file and extract health-results.json file
+			tar -xzf $outFile -C $outFilePath | Out-NULL
+			$healthSummaryPath = gci -recurse -filter "health-results.json" -Path $outFilePath
+			$healthSummaryFile = $healthSummaryPath.DirectoryName + "\health-results.json"
+			Copy-Item $healthSummaryFile $reportDestination  | Out-NULL
+				
+			# Remove Temporary Directory
+			Remove-Item -Recurse -Force $outFilePath  | Out-NULL
+				
+            # Converting to JSON
+            $temp = Get-Content -Path $reportDestination; $temp = $temp -replace '""', '"-"'; $temp | Out-File $reportDestination
+			return $reportDestination
```

Comment:
```suggestion
			Return $reportDestination
```

Reply:
addressed in commit d64ce9ff50fe306dc3219c3013b0849ffc2e49c5